src/gridcoin/upgrade.cpp

```diff
@@ -135,7 +136,14 @@ bool Upgrade::CheckForLatestUpdate(bool ui_dialog, std::string client_message_ou
                 break;
 
             if (std::stoi(GithubVersion[x]) > LocalVersion[x])
+            {
                 NewVersion = true;
+
+                if (x < 2)
+                {
+                    NewMandatory = true;
+                }
+            }
```

Comment:
Is it possible to use something like the (unlocalized) value of `GithubReleaseType` here instead of the version number? Using _minor_ as in _major.minor.maintenance_&mdash;even though we don't follow semantic versioning&mdash;may cramp the ability to choose version numbers for non-mandatory releases.

Reply:
I am not sure I understand your comment. Mandatory releases have always been defined as a change in the first or second of the quad, while the third is a leisure, and the fourth is non-production (always zero for production). Do we need any more flexibility than that?

Reply:
On second thought, I don't think what you are suggesting is going to work... This is specifically meant to deal with the situation where the GitHubRelease Type is leisure, but the person's wallet is below the last mandatory. Using the GitHubReleaseType for the current release doesn't help you that much. Clearly if it was a mandatory, problem already solved (even without this PR....)

To not depend on the encoding of leisure via the triple to distinquish this case would require us to pull iteratively every release until we get to the last mandatory. I think that is too much.